riffdog/scanner.py

```diff
@@ -5,12 +5,17 @@
 import pkgutil
 from importlib import import_module
 
-import boto3
-
 from .config import RDConfig, StateStorage
 from .resource import ResourceDirectory
 from .exceptions import ResourceNotFoundError, StorageNotImplemented
 
+try:
+    import boto3
```

Comment:
Why would you not have both installed - it's a requirement of installing riffdog

Reply:
So boto3 at the moment is a requirement (as in its explicit in the setup.py), but it doesn't have to be. Now we have file support, this is now effectively optional - if you're not using AWS (for example, let's say you're using file based support, and want to use `riffdog_cloudflare`, then you don't want to install boto.

What I havn't worked out is quite how this affects the installation to get new users using it faster, so at the *moment* i think it makes sense to leave boto in the setup.py requirements.

(basically, `pip install riffdog[aws]` installs the aws module, so you would have to, if you wanted to do cloudflare with an s3 state file, is have a requirement of like `s3_states` and do something like `pip install riffdog[cloudflare,s3_states]` which is less obvious. - you wouldn't need to do that for aws + s3 file share as aws installs boto...... I don't see an obvious solution right now, but I want the code to be ready for it).

Also this line is difficult to test - Tox lets it run different environments, and I think theres an env var we can access to tell us we're in Tox, so we might be able to make a tox-environment thing that has boto, and one that doesn't, and that then skips test (or just passes them or something...).

Reply:
I don't see what a big deal it is to just have boto as a requirement - if people use riffdog and complain about it then we could readdress it but at this point I don't see it as being a big deal.

Reply:
Okay, - I'm not removing teh support yet (that I fully agree). but I'm going to leave this if in for the moment?

Reply:
Sure, no problem